packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -109,76 +54,430 @@ const invalidateCustomerBasketsQuery = (
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                {queryKey: getTaxesFromBasket.queryKey(parameters)}
+            ]
+        }
+    },
+    addTaxesForBasketItem(customerId, {parameters}) {
+        return {
+            // TODO: Convert invalidate to an update that removes the matching basket
+            ...invalidateCustomerBasketsQuery(customerId, parameters),
+            update: [{queryKey: getBasket.queryKey(parameters)}]
+        }
+    },
+    createBasket(customerId, {parameters}, response) {
+        const {basketId} = response
+
+        return {
+            update: [
+                {queryKey: getBasket.queryKey({...parameters, basketId})},
+                ...(customerId && basketId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({
+                                  ...parameters,
+                                  customerId
+                              }),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(
+                                      {...parameters, basketId},
+                                      response,
+                                      oldData
+                                  )
+                          }
+                      ]
+                    : [])
+            ],
+            ...(!basketId && invalidateCustomerBasketsQuery(customerId, parameters))
+        }
+    },
+    createShipmentForBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    deleteBasket(customerId, {parameters}) {
+        return {
+            // TODO: Convert invalidate to an update that removes the matching basket
+            ...invalidateCustomerBasketsQuery(customerId, parameters),
+            remove: [
+                // We want to fuzzy match all queryKeys with the `basketId` in their path
+                // `["/organizations/",${organization},"/baskets/",${basketId}]`
+                {queryKey: getBasket.path(parameters)}
+            ]
+        }
+    },
+    mergeBasket(customerId, {parameters}, response) {
+        const {basketId} = response
+        return {
+            update: [
+                {queryKey: getBasket.queryKey({...parameters, basketId})},
+                ...(customerId && basketId
```

Comment:
I think this one also needs the fallback to invalidate if `basketId` is missing.